src/Health/Checks/ClientConnectivityHealthCheck.php

```diff
@@ -0,0 +1,124 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Health\Checks;
+
+use EoneoPay\Externals\DataTransferObjects\Health\HealthState;
+use EoneoPay\Externals\Health\Interfaces\HealthCheckInterface;
+use EoneoPay\Externals\Health\Interfaces\HealthInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+
+/**
+ * This health check is used to check that a connection to a guzzle client can be made.
+ * Details of response is not intended to be checked, but to make sure a connection attempt
+ * does not throw exception.
+ *
+ * The endpoint to connect to must always be a GET.
+ */
+class ClientConnectivityHealthCheck implements HealthCheckInterface
+{
+    /**
+     * The degraded state message.
+     *
+     * @const string
+     */
+    private const MESSAGE_DEGRADED = 'The connection to client is degraded.';
+
+    /**
+     * The healthy state message.
+     *
+     * @const string
+     */
+    private const MESSAGE_HEALTHY = 'The connection to client is healthy.';
+
+    /**
+     * @var \GuzzleHttp\ClientInterface
+     */
+    private $client;
+
+    /**
+     * Endpoint to connect to.
+     *
+     * @var string
+     */
+    private $endpoint;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * ClientConnectivityHealthCheck constructor.
+     *
+     * @param \GuzzleHttp\ClientInterface $client
+     * @param string $endpoint Endpoint to hit. This needs to be a GET method.
+     * @param string $serviceName The name of service you are connecting to. Used when displaying service health.
+     */
+    public function __construct(ClientInterface $client, string $endpoint, string $serviceName)
```

Comment:
I am thinking of this specific health check as a generic check which can be used to test for multiple external clients. Thus passing in all these arguments to be able to configure it.